### ISSUE TITLE
Ready function of child called after entire adding function is done, if parent node is not ready.

### ISSUE THREAD

**Issue by UltyX** [godotengine/godot#7828]
Change in _ready behavior from 2.1.1 to 2.1.2

**Issue description:**
Strange behavior when adding a node to a node that is not ready yet.
Adding a child to a node, which is not yet ready, triggers the ready function of the child only after the  function, which is adding the child, is done.
The issue here would be an unexpected break in sequence, since normally the ready function is called right after the add_child( new_child_instance ) is done. 

**Link to minimal example project:**

In this small example it is shown that adding a child node to a not yet ready node gives a different sequence (add_child(), print(), child._ready() ) to when the same is done to a node which is ready (add_child(), child._ready(), print() ).

[ready.zip](https://github.com/godotengine/godot/files/783503/ready.zip)

**Comment by kubecz3k** [comment]
In 3.0 I don't think this behavior is reproducible anymore since you will have an error if you will try to add new child to not ready yet node in 3.0.
The original issue however was about behavior change between 2.1.1 and 2.1.2 not sure what's the status here.

**Comment by UltyX** [comment]
I just imported and tested the example with 3.0.2 and the behavior is still reproducible. Here is the updated example for 3.0
[ready_3_0.zip](https://github.com/godotengine/godot/files/1906904/ready_3_0.zip)

**Comment by UltyX** [comment]
I think this issue is no longer relevant since adding a child in this manner should be done with a call deferred anyway.
